python/flashinfer/utils.py

```diff
@@ -202,26 +205,46 @@ def _check_cached_qkv_data_type(
         )
 
 
-def register_custom_op(
-    name: str,
-    fn: Optional[Callable] = None,
-    /,
-    *,
-    mutates_args: Union[str, Iterable[str]],
-    device_types: Optional[Union[str, Sequence[str]]] = None,
-    schema: Optional[str] = None,
-) -> Callable:
-    if TorchVersion(torch_version) < TorchVersion("2.4"):
-        return lambda x: x
-    return torch.library.custom_op(
-        name, fn, mutates_args=mutates_args, device_types=device_types, schema=schema
-    )
+if IS_BUILDING_DOCS or TorchVersion(torch_version) < TorchVersion("2.4"):
```

Comment:
QQ Why do we need to place `register_custom_op` below this `if` statement

Reply:
We need to define this two functions anyways. They are used to annotate the kernels. When the real implementation does not exist, these two functions return an identity function. So the kernels being annotated will remain unchanged after applying the annotation.

Reply:
Oh, do you mean why not put this `if` inside `register_custom_op`? Basically, I'm trying to avoid checking this condition every time this function is called. The condition is constant for the whole program's life.

Either approach is fine. The annotation is applied to kernels when we define them. These two functions won't be invoked when we call the kernels.